src/operations/extension/reduction_partial_rows.hpp

```diff
@@ -0,0 +1,237 @@
+  static constexpr index_t work_group_rows = params_t::work_group_rows;
+  static constexpr index_t work_group_cols = params_t::work_group_cols;
+  static constexpr index_t local_memory_rows = params_t::local_memory_rows;
+  static constexpr index_t local_memory_cols = params_t::local_memory_cols;
+  static constexpr index_t local_memory_size = params_t::local_memory_size;
+
+  /* Neutral value for this reduction operator */
+  static constexpr value_t init_val = operator_t::template init<output_t>();
+
+  /* Input and output buffers */
+  input_t in_;
+  output_t out_;
+
+  /* Matrix dimensions */
+  const index_t rows_;
+  const index_t cols_;
+  const index_t leading_dim_;
+
+  /* Work groups per dimension */
+  const index_t group_count_rows_;
+  const index_t group_count_cols_;
+
+  SYCL_BLAS_INLINE ReductionPartialRows(input_t in, output_t out,
+                                        index_t group_count_cols)
+      : in_(in),
+        out_(out),
+        rows_(in_.get_size_row()),
+        cols_(in_.get_size_col()),
+        leading_dim_(in_.getSizeL()),
+        group_count_rows_((rows_ - 1) / local_memory_rows + 1),
+        group_count_cols_(group_count_cols) {}
+
+  void bind(cl::sycl::handler &h) {
+    in_.bind(h);
+    out_.bind(h);
+  }
+  void adjust_access_displacement() {
+    in_.adjust_access_displacement();
+    out_.adjust_access_displacement();
+  }
+  SYCL_BLAS_INLINE bool valid_thread(cl::sycl::nd_item<1> ndItem) const {
+    return true;
+  }
+
+  /*!
+   * @brief get_workgroup_cluster. This function is used to find the optimum
+   * number of work groups required to execute each partial reduction step.
+   */
+  SYCL_BLAS_INLINE index_t get_workgroup_cluster() noexcept {
+    return ((rows_ - 1) / local_memory_rows + 1) * group_count_cols_;
+  }
+  /*!
+   * @brief get_num_workgroup_cluster. This function is used to extend the
+   * number of work group clusters, in order to make sure that at least 4
+   * operations are available per work group. The number 4 is used based on
+   * empirical research.
+   */
+  SYCL_BLAS_INLINE index_t
+  get_num_workgroup_cluster(index_t compute_units) noexcept {
+    return ((4 * compute_units - 1) / get_workgroup_cluster() + 1);
+  }
+
+  /*!
+   * @brief Get the nd_range value which has to be used for kernels that
+   *        intend to call GemmPartial::run().
+   */
+  SYCL_BLAS_INLINE cl::sycl::nd_range<1> get_nd_range(
+      index_t compute_units) noexcept {
+    const cl::sycl::range<1> nwg(get_workgroup_cluster() *
+                                 get_num_workgroup_cluster(compute_units));
+    const cl::sycl::range<1> wgs(work_group_size);
+    return cl::sycl::nd_range<1>(nwg * wgs, wgs);
+  }
+
+  template <typename local_memory_t>
+  SYCL_BLAS_INLINE void eval(local_memory_t scratch,
+                             cl::sycl::nd_item<1> id) noexcept {
+    /* reference to the scratch memory */
+    auto scratch_ptr = scratch.localAcc.get_pointer().get();
```

Comment:
I've updated the line where the pointer is obtained.
Concerning bank conflicts, the only case where it can be an issue is for a very small tile at the last steps of the parallel reduction. Even in that rare case it would not affect the performance significantly